builder/parser.nix

```diff
@@ -0,0 +1,134 @@
+# Parse go.mod in Nix
+# Returns a Nix structure with the contents of the go.mod passed in
+# in normalised form.
+
+let
+  inherit (builtins) elemAt mapAttrs split foldl' match filter typeOf;
+
+  # Strip lines with comments & other junk
+  stripStr = s: elemAt (split "^ *" (elemAt (split " *$" s) 0)) 2;
+  stripLines = initialLines: foldl' (acc: f: f acc) initialLines [
+    # Strip comments
+    (lines: map
+      (l:
+        let
+          m = match "(.*)( |)//.*" l;
+          hasComment = m != null;
+        in
+        stripStr (if hasComment then elemAt m 0 else l))
+      lines)
+
+    # Strip leading tabs characters
+    (lines: map (l: elemAt (match "(\t|)(.*)" l) 1) lines)
```

Comment:
it complains `(\t|)(.*)` is invalid regex.

Reply:
Are you on MacOS by any chance?

Reply:
> Are you on MacOS by any chance?

yes

Reply:
@adisbladis What's the meaning of `|` in this context, I only know it can represent `or`, but I guess it's not like that here.